lib/services/api/api_service.dart

```diff
@@ -53,40 +131,46 @@ class ApiServiceImpl implements ApiService {
       headers["authorization"] = "$finalTokenType $finalToken";
     }
 
-    final url = finalBaseUrl! + endPoint;
+    final url = finalBaseUrl! + endpoint;
     try {
-      final response = await _httpService.request(
+      return await _httpService.request(
         method: method,
         data: params?.toJson(),
         url: url,
         headers: headers,
       ) as Map<String, dynamic>;
-      return Mapper.fromJson(
-              response["data"]["attributes"] as Map<String, dynamic>)
-          .toObject<T>();
     } on HttpException catch (e) {
       throw ApiException.fromHttpException(e) ?? e;
     }
   }
 
-  /// Configures a default base url when [call] with `baseUrl` is null.
-  ///
-  /// This value is stored in a static variable [_baseUrl] so it will be available in any
-  /// instance of [ApiServiceImpl], after set
-  @override
-  void configureGlobalBaseUrl(String? baseUrl) {
-    _baseUrl = baseUrl;
+  T _convertResponseToObject<T>(Map<String, dynamic> response) {
+    if (response["data"] == null && T.toString() == "void") {
+      return null as T;
+    }
+
+    if (response["data"] is! Map<String, dynamic> ||
+        response["data"]["attributes"] is! Map<String, dynamic>) {
+      throw ApiException.wrongResponseStructure;
+    }
+
+    return Mapper.fromJson(
+            response["data"]["attributes"] as Map<String, dynamic>)
+        .toObject<T>()!;
   }
 
-  /// Configures a default token & tokenType when [call] with token or tokenType is null.
-  ///
-  /// This value is stored in a static variable [_token], [_tokenType] so it will be available in any
-  /// instance of [ApiServiceImpl], after set
-  @override
-  void configureGlobalToken(String? token, String? tokenType) {
-    _token = token;
-    if (tokenType != null) {
-      _tokenType = tokenType;
+  ApiListObject<T> _convertResponseToListObject<T>(
+      Map<String, dynamic> response) {
+    if (response["data"] is! List<dynamic>) {
+      throw ApiException.wrongResponseStructure;
     }
+
+    final items = (response["data"] as List<dynamic>)
+        .where((e) => e["attributes"] is Map<String, dynamic>)
```

Comment:
not sure if this is useful as I saw you're doing the typecasting below as well 🤔  is it necessary?

Reply:
Just make sure only Map go thru,
The casting below will crash if there are a element doesn't.
😄